systemcontractindex/stakingindex/bucket_test.go

```diff
@@ -0,0 +1,156 @@
+package stakingindex
+
+import (
+	"math/big"
+	"math/rand/v2"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/iotexproject/iotex-address/address"
+
+	"github.com/iotexproject/iotex-core/v2/test/identityset"
+)
+
+func TestBucket_SerializeDeserialize(t *testing.T) {
+	r := require.New(t)
+	candidate := identityset.Address(1)
+	owner := identityset.Address(2)
+
+	original := &Bucket{
+		Candidate:      candidate,
+		Owner:          owner,
+		StakedAmount:   big.NewInt(123456),
+		Timestamped:    true,
+		StakedDuration: 3600,
+		CreatedAt:      111111,
+		UnlockedAt:     222222,
+		UnstakedAt:     333333,
+		Muted:          true,
+	}
+	data := original.Serialize()
+
+	var deserialized Bucket
+	r.NoError(deserialized.Deserialize(data), "Deserialize failed")
+	r.Equal(*original, deserialized)
+}
+
+func TestBucket_Clone(t *testing.T) {
+	r := require.New(t)
+	candidate := identityset.Address(1)
+	owner := identityset.Address(2)
+	original := &Bucket{
+		Candidate:      candidate,
+		Owner:          owner,
+		StakedAmount:   big.NewInt(123456),
+		Muted:          true,
+		Timestamped:    true,
+		StakedDuration: 3600,
+		CreatedAt:      111111,
+		UnlockedAt:     222222,
+		UnstakedAt:     333333,
+	}
+
+	clone := original.Clone()
+	r.Equal(*original, *clone, "Clone failed")
+}
+
+func TestAssembleVoteBucket(t *testing.T) {
+	r := require.New(t)
+	mockBlocksToDuration := func(height, end uint64) time.Duration {
+		return time.Duration(end-height) * time.Second
+	}
+	candidate := identityset.Address(1)
+	owner := identityset.Address(2)
+	contract := identityset.Address(3)
+
+	cases := []struct {
+		name   string
+		bucket *Bucket
+	}{
+		{"timestamped", &Bucket{
+			Candidate:      candidate,
+			Owner:          owner,
+			StakedAmount:   big.NewInt(1000),
+			Timestamped:    true,
+			StakedDuration: 3600,
+			CreatedAt:      111111,
+			UnlockedAt:     222222,
+			UnstakedAt:     333333,
+			Muted:          false,
+		}},
+		{"block-based", &Bucket{
+			Candidate:      candidate,
+			Owner:          owner,
+			StakedAmount:   big.NewInt(2000),
+			Timestamped:    false,
+			StakedDuration: 100,
+			CreatedAt:      10,
+			UnlockedAt:     20,
+			UnstakedAt:     110,
+			Muted:          false,
+		}},
+		{"block-based/muted", &Bucket{
+			Candidate:      candidate,
+			Owner:          owner,
+			StakedAmount:   big.NewInt(2000),
+			Timestamped:    false,
+			StakedDuration: 100,
+			CreatedAt:      10,
+			UnlockedAt:     20,
+			UnstakedAt:     110,
+			Muted:          true,
```

Comment:
is it possible both `Muted` and `Timestamped` are true? if that's possible add a test case to cover that